api/tests/integration/domain/usecases/create-password-reset-demand_test.js

```diff
@@ -0,0 +1,64 @@
+const { catchErr, databaseBuilder, expect, knex } = require('../../../test-helper');
+
+const mailService = require('../../../../lib/domain/services/mail-service');
+const resetPasswordService = require('../../../../lib/domain/services/reset-password-service');
+const authenticationMethodRepository = require('../../../../lib/infrastructure/repositories/authentication-method-repository');
+const resetPasswordDemandRepository = require('../../../../lib/infrastructure/repositories/reset-password-demands-repository');
+const userRepository = require('../../../../lib/infrastructure/repositories/user-repository');
+
+const { UserNotFoundError } = require('../../../../lib/domain/errors');
+
+const createPasswordResetDemand = require('../../../../lib/domain/usecases/create-password-reset-demand');
+
+describe('Integration | UseCases | create-password-reset-demand', () => {
+
+  const email = 'user@example.net';
+  const locale = 'fr';
+
+  beforeEach(async () => {
+    const userId = databaseBuilder.factory.buildUser({ email }).id;
+    databaseBuilder.factory.buildAuthenticationMethod.buildWithHashedPassword({ userId });
+    await databaseBuilder.commit();
+  });
+
+  afterEach(async () => {
+    await knex('reset-password-demands').delete();
+  });
+
+  it('should create a password reset demand', async () => {
+    // when
+    const result = await createPasswordResetDemand({
+      email,
+      locale,
+      mailService,
+      resetPasswordService,
+      authenticationMethodRepository,
+      resetPasswordDemandRepository,
+      userRepository,
+    });
+
+    // then
+    expect(result.attributes.email).to.deep.equal(email);
+    expect(result.attributes.temporaryKey).to.be.ok;
+  });
+
+  it('should throw UserNotFoundError if user email does not exist', async () => {
```

Comment:
**chipotage**
`it('should throw UserNotFoundError if no user account exists with this email', async () => {`
pour éviter qu'on pense que l'adresse email n'existe pas (au sens SMTP)

Reply:
Done